### DISCUSSION TITLE
Random stock for traders

### DISCUSSION THREAD

**Discussion by Danimal696** [flareteam/flare-game#877] [Q&A] [answered]
Would it be possible to declare a range of goods to be randomly shown at a traders shop?. Something like declaring:
random_item(100 - 200):1 (1 item randomly picked between 100 to 200)

**Comment by dorkster** [comment, marked as answer]
Have you tried using the `random_stock` and `random_stock_count` properties? They let you use a loot table as the source of items sold by a vendor. For example, Abasi in the Empyrean campaign has the following:

```
random_stock=loot/trader_abasi.txt
random_stock_count=10,12
```

```
### loot/trader_abasi.txt

# normal clothes
loot=4,100
loot=5,100
loot=6,100
loot=7,100

# normal weapons
loot=8,100
loot=9,100
loot=10,100

# normal misc
loot=11,100

# magic clothes
loot=12,75
loot=13,75
loot=14,75
loot=15,75
loot=16,75
loot=17,75
loot=18,75
loot=19,75
loot=20,75
loot=21,75
loot=22,75
loot=23,75
loot=24,75
loot=25,75
loot=26,75

# magic weapons
loot=27,75
loot=28,75
loot=29,75

# magic misc
loot=30,75
loot=31,75
loot=32,75
loot=33,75
loot=34,75
loot=35,75
loot=36,75
loot=37,75

# epic clothes
loot=38,25
loot=39,25
loot=40,25
loot=41,25
loot=42,25
loot=43,25
loot=44,25
loot=45,25
loot=46,25
loot=47,25
loot=48,25
loot=49,25
loot=50,25
loot=51,25
loot=52,25

# epic weapons
loot=53,25
loot=54,25
loot=55,25

# epic misc
loot=56,25
loot=57,25
loot=58,25
loot=59,25
loot=60,25
loot=61,25
loot=62,25
loot=63,25
```

**Comment by Danimal696** [comment]
No, i didn´t know of that command. Thanks for the answer, this will really spice up the seller.

**Comment by Danimal696** [comment]
Another one. Can i set an individual seller price higher? I know you can scale potion prices by level and global prices by "vendor_ratio". Can it be done for just a single elite items seller? So he sells rare or unique items randomly selected but at 300% - 500% normal price.

**Reply by dorkster** [reply]
Not at the moment, but I think a per-vendor setting for price scaling is a good idea.

**Comment by Danimal696** [comment]
Following on this question, can "status_stock" affect "random_stock"? So random items of a certain range appear only if a status has been set.

**Reply by dorkster** [reply]
Not at the moment. I think a good solution to this would be to add support for checking status to loot tables. We use EventComponent there as a way to store item ID/quantity/chance, and the status field is currently unused in that context.

**Comment by Danimal696** [comment]
Pity, it would be a nice addition having the trader expands his wares with higer qualities random items as story progress.
